### ISSUE TITLE
[Bug]: cssnano transformations are unsafe in custom property values

### ISSUE THREAD

**Issue by nex3** [cssnano/cssnano#1456]
### Describe the bug

In cssnano compilation modes that seek to guarantee exact behavioral equivalence between the input and output CSS, transformations on custom property values should be substantially limited relative to what they are now. Custom properties aren't just used for configuring values elsewhere in the stylesheet; they may also be used to communicate information from the stylesheet to JavaScript (as in Polymer.js for example), and so the details of their JS serialization is semantically relevant. cssnano currently modifies these values like any other property, which can change the serialization and thus the semantics of the stylesheet.

There are two broad categories of issue here:

1. The existence of whitespace tokens is visible to JS serialization, so `--foo: bar` and `--foo:bar` are meaningfully different. cssnano will convert the former to the latter.

2. There's no guarantee that a custom property's value will be parsed in a context where a CSS color is meaningful, so `--foo: rgb(0 0 0)` and `--foo: #000` are meaningfully different. cssnano will convert the former to the latter.

The second issue covers essentially all value-level cssnano optimizations, almost none of which are safe to perform on a custom property value.

### Expected behaviour

There should at least be a way to tell cssnano not to make changes that could affect the semantics of the compiled CSS—which is to say, to disable essentially all value-level conversions for custom properties unless they don't affect the way those properties are serialized to JS values. Ideally, this configuration would be enabled in `cssnano-preset-default`, since that advertises itself as making no assumptions about how the CSS will be used.

### Steps to reproduce

Copy/paste this into https://cssnano.co/playground/:

```css
a {
  --foo: rgb(0 0 0);
}
```

### Version

5.1.14

### Preset

default

### Environment

```shell
cssnano playground
```

### Package details

```shell
cssnano playground
```

**Comment by alexander-akait** [comment]
Yeah, it is unsafe, we should provide an option to disable it, I want to keep it `true` by default, because most of developer use CSS inside custom properties, so it is safe in most of cases

**Comment by nex3** [comment]
That's fair, but in that case you should probably change the description of `cssnano-prefix-default` to avoid saying it makes no assumptions. (Probably just enumerating the assumptions it _does_ make is sufficient.)

**Comment by ludofischer** [comment]
> The existence of whitespace tokens is visible to JS serialization, so --foo: bar and --foo:bar are meaningfully different. cssnano will convert the former to the latter.

This part is not solved yet
